release-tool.ps1

```diff
@@ -0,0 +1,456 @@
+    [string] $SignKey,
+    [Parameter(ParameterSetName = "build")]
+    [Parameter(ParameterSetName = "sign")]
+    [string] $Timestamp = "http://timestamp.sectigo.com",
+    [Parameter(ParameterSetName = "merge")]
+    [Parameter(ParameterSetName = "build")]
+    [Parameter(ParameterSetName = "sign")]
+    [string] $GpgKey = "CFB4C2166397D0D2",
+    [Parameter(ParameterSetName = "merge")]
+    [Parameter(ParameterSetName = "build")]
+    [string] $SourceDir = ".",
+    [Parameter(ParameterSetName = "build")]
+    [string] $OutDir = ".\release",
+    [Parameter(ParameterSetName = "merge")]
+    [Parameter(ParameterSetName = "build")]
+    [string] $Tag,
+    [Parameter(ParameterSetName = "merge")]
+    [string] $SourceBranch,
+    [Parameter(ParameterSetName = "merge")]
+    [string] $TargetBranch = "master",
+    [Parameter(ParameterSetName = "build")]
+    [string] $VSToolChain,
+    [Parameter(ParameterSetName = "merge")]
+    [Parameter(ParameterSetName = "build")]
+    [Parameter(ParameterSetName = "sign")]
+    [string] $ExtraPath
+)
+
+# Helper function definitions
+function Test-RequiredPrograms {
+    # If any of these fail they will throw an exception terminating the script
+    if ($Build) {
+        Get-Command git | Out-Null
+        Get-Command cmake | Out-Null
+    }
+    if ($Merge) {
+        Get-Command git | Out-Null
+        Get-Command tx | Out-Null
+        Get-Command lupdate | Out-Null
+    }
+    if ($Sign -or $SignBuild) {
+        if ($SignKey.Length) {
+            Get-Command signtool | Out-Null
+        }
+        Get-Command gpg | Out-Null
+    }
+}
+
+function Test-VersionInFiles {
+    # Check CMakeLists.txt
+    $Major, $Minor, $Patch = $Version.split(".", 3)
+    if (-not (Select-String "$SourceDir\CMakeLists.txt" -pattern "KEEPASSXC_VERSION_MAJOR `"$Major`"" -Quiet) `
+            || -not (Select-String "$SourceDir\CMakeLists.txt" -pattern "KEEPASSXC_VERSION_MINOR `"$Minor`"" -Quiet) `
+            || -not (Select-String "$SourceDir\CMakeLists.txt" -pattern "KEEPASSXC_VERSION_PATCH `"$Patch`"" -Quiet)) {
+        throw "CMakeLists.txt has not been updated to $Version."
+    }
+
+    # Check Changelog
+    if (-not (Select-String "$SourceDir\CHANGELOG.md" -pattern "^## $Version \(\d{4}-\d{2}-\d{2}\)$" -Quiet)) {
+        throw "CHANGELOG.md does not contain a section for $Version."
+    }
+
+    # Check AppStreamInfo
+    if (-not (Select-String "$SourceDir\share\linux\org.keepassxc.KeePassXC.appdata.xml" `
+                -pattern "<release version=`"$Version`" date=`"\d{4}-\d{2}-\d{2}`">" -Quiet)) {
+        throw "share/linux/org.keepassxc.KeePassXC.appdata.xml does not contain a section for $Version."
+    }
+
+    # Check Snapcraft
+    if (-not (Select-String "$SourceDir\snap\snapcraft.yaml" -pattern "version: $Version" -Quiet)) {
+        throw "snap/snapcraft.yaml has not been updated to $Version."
+    }
+}
+
+function Test-WorkingTreeClean {
+    $(git diff-index --quiet HEAD --)
```

Comment:
I'm wondering if that is really the correct syntax. At least in Bash, this syntax is used only for capturing the output value. Otherwise you would simply use normal parentheses without the dollar sign or nothing at all if you don't need a sub shell. For PowerShell, I've also seen a `&` syntax, which seems to be largely equivalent to `InvokeExpression`.